src/main/java/org/generationcp/middleware/dao/SampleDao.java

```diff
@@ -527,29 +534,39 @@ private void appendSamplesFromQuery(final StringBuilder sql) {
 	}
 
 	private void appendSampleSeachFilters(final StringBuilder sql, final SampleSearchRequestDTO requestDTO) {
-		if(!StringUtil.isEmpty(requestDTO.getSampleDbId())) {
-			sql.append(" AND s.sample_bk = :sampleDbId");
+		if(!CollectionUtils.isEmpty(requestDTO.getSampleDbIds())) {
+			sql.append(" AND s.sample_bk in (:sampleDbIds)");
 		}
-		if(!StringUtil.isEmpty(requestDTO.getObservationUnitDbId())) {
-			sql.append(" AND e.obs_unit_id = :observationUnitDbId");
+		if(!CollectionUtils.isEmpty(requestDTO.getObservationUnitDbIds())) {
+			sql.append(" AND e.obs_unit_id in (:observationUnitDbIds)");
 		}
-		if(!StringUtil.isEmpty(requestDTO.getPlateDbId())) {
-			sql.append(" AND s.plate_id = :plateDbId");
+		if(!CollectionUtils.isEmpty(requestDTO.getPlateDbIds())) {
+			sql.append(" AND s.plate_id in (:plateDbIds)");
 		}
-		if(!StringUtil.isEmpty(requestDTO.getGermplasmDbId())) {
-			sql.append(" AND g.germplsm_uuid = :germplasmDbId");
+		if(!CollectionUtils.isEmpty(requestDTO.getGermplasmDbIds())) {
+			sql.append(" AND g.germplsm_uuid in (:germplasmDbIds)");
 		}
-		if(!StringUtil.isEmpty(requestDTO.getStudyDbId())) {
-			sql.append(" AND e.nd_geolocation_id = :studyDbId");
+		if(!CollectionUtils.isEmpty(requestDTO.getStudyDbIds())) {
+			sql.append(" AND e.nd_geolocation_id in (:studyDbIds)");
 		}
-		if (!StringUtil.isEmpty(requestDTO.getExternalReferenceID())) {
+		if (!CollectionUtils.isEmpty(requestDTO.getExternalReferenceIDs())) {
 			sql.append(" AND EXISTS (SELECT * FROM external_reference_sample sref ");
-			sql.append(" WHERE s.sample_id = sref.sample_id AND sref.reference_id = :referenceId) ");
+			sql.append(" WHERE s.sample_id = sref.sample_id AND sref.reference_id in (:referenceIds)) ");
 		}
 
-		if (!StringUtil.isEmpty(requestDTO.getExternalReferenceSource())) {
+		if (!CollectionUtils.isEmpty(requestDTO.getExternalReferenceSources())) {
 			sql.append(" AND EXISTS (SELECT * FROM external_reference_sample sref ");
-			sql.append(" WHERE s.sample_id = sref.sample_id AND sref.reference_source = :referenceSource) ");
+			sql.append(" WHERE s.sample_id = sref.sample_id AND sref.reference_source in (:referenceSources)) ");
+		}
+
+		// Search preferred names
+		if (!CollectionUtils.isEmpty(requestDTO.getGermplasmNames())) {
+			sql.append(" AND g.gid IN ( SELECT n.gid ");
+			sql.append(" FROM names n WHERE n.nstat = 1 AND n.nval in (:germplasmNames) ) ");
+		}
+
+		if (!CollectionUtils.isEmpty(requestDTO.getStudyNames())) {
+			sql.append(" AND CONCAT(pmain.name, ' Environment Number ', geoloc.description) in (:studyNames)");
```

Comment:
Can you extract this as constant (maybe `STUDY_NAME_BRAPI`) in `DmsProjectDao` and use it here:
`CONCAT(pmain.name, ' Environment Number ', geoloc.description)`